cmd/pj-rehearse/main.go

```diff
@@ -168,14 +168,34 @@ func main() {
 		changedCiopConfigs = diffs.GetChangedCiopConfigs(masterConfig.CiOperator, prConfig.CiOperator, logger)
 	}
 
+	changedTemplates := make(config.CiTemplates)
 	// We can only detect changes if we managed to load both CI template versions
 	if masterConfig.Templates != nil && prConfig.Templates != nil {
-		changedTemplates := diffs.GetChangedTemplates(masterConfig.Templates, prConfig.Templates, logger)
+		changedTemplates = diffs.GetChangedTemplates(masterConfig.Templates, prConfig.Templates, logger)
 		for name := range changedTemplates {
 			logger.WithField("template-name", name).Info("Changed template")
 		}
 	}
 
+	cmClient, err := rehearse.NewCMClient(clusterConfig, masterConfig.Prow.ProwJobNamespace, o.dryRun)
+	if err != nil {
+		logger.WithError(err).Error("could not create a configMap client")
+		gracefulExit(o.noFail, misconfigurationOutput)
+	}
+
+	cmController := config.NewTemplateCMController(cmClient, prNumber, logger, changedTemplates)
+	if err := cmController.CreateCMTemplates(); err != nil {
+		logger.WithError(err).Error("couldn't create template configMap")
+		gracefulExit(o.noFail, "")
```

Comment:
Two things:

1. Do not pass an empty string as a message that's intended to be user-facing
2. This will still leave some CMs behind if `CreateCMTemplates` manage to create some and then fails.